tests/integration/conftest.py

```diff
@@ -21,9 +18,43 @@ def copy_loki_library_into_tester_charm(ops_test):
     shutil.copyfile(library_path, install_path)
 
 
+@pytest.fixture(scope="module")
+async def loki_charm(ops_test: OpsTest):
+    """Loki charm used for integration testing."""
+    charm = await ops_test.build_charm(".")
+    return charm
+
+
 @pytest.fixture(scope="module")
 async def loki_tester_charm(ops_test):
     """A charm for integration test of the Loki charm."""
     charm_path = "tests/integration/loki-tester"
+    clean_cmd = ["charmcraft", "clean", "-p", charm_path]
+    await ops_test.run(*clean_cmd)
+    bad_rule_path = "tests/integration/loki-tester/src/loki_alert_rules/free-standing/error.rule"
+    try:
+        os.remove(bad_rule_path)
+    except FileNotFoundError:
+        pass
     charm = await ops_test.build_charm(charm_path)
     return charm
+
+
+@pytest.fixture(scope="module")
+async def faulty_loki_tester_charm(ops_test):
```

Comment:
I understand that this was added for resiliency, but this adds an enormous amount of runtime to the integration tests simply to avoid `cos-config`.

As long as `cos-config` "points" at a non-`main` branch here, we can avoid the massive runtime hit from completely cleaning and rebuilding both charms on every test.

Reply:
This is an issue with charmcraft. A `charmcraft clean` will not be necessary once charmcraft caching is more intelligent.

Reply:
Frankly, "this is an issue with something else" is both true and meaningless. 

"Unless the opposition to using rules in `cos-config` for the single error check which would avoid this entire clean/repack thing is something other than philosophical, we are cutting of our nose to spite our face.

A `charmcraft clean` would also not be necessary if the error rules check pulled from a different branch (which I recreated so it passed in my PR). Using `loki-tester` is great. It does not have to be this prescriptive.

Reply:
One lesson I would draw from item 1 in the PR description is when ever possible if we can use test doubles and charm local tests we should.